### PR TITLE
chore(package): add `patch-package` and apply patch

### PR DESCRIPTION
**Pull request by adnasa**
#### Summary

- adds `patch-package` and apply patch on `@types/react-textarea-autosize` until the official package is updated.

## Description

This is an intermediate solution until https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52391 is merged.
Most of the diff on the patch file is due to indentation, but the important changes are:

1. updating signature of `onHeightChange`
2. exposing `TextareaHeightChangeMeta` (for optional use)
3. exposing `style` prop as it is defined on the original package (https://github.com/Andarist/react-textarea-autosize/blob/master/src/index.tsx#L24:L24)

## Open question

The `patch-package` has recommendations on updating the CI. not sure how to translate that into GH actions. I welcome pairing on this.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   "keywords": ["javascript", "frontend", "ui-kit", "design-system", "react"],
   "sideEffects": false,
   "scripts": {
-    "postinstall": "manypkg check && preconstruct dev && yarn --cwd docs",
+    "postinstall": "patch-package",
```
Comment (adnasa) [inline package.json:16]: oh wait..

**Comment by tdeekens** [comment]
Can we remove the indentation somehow? I wonder if it causes issues if the library moves in for us unrelated/uninteresting lines.

**Review by emmenko — COMMENTED** [review]
I'm confused. Doesn't the library have it's own types?
Why do we need the outdated types package? 🤔

**Comment by adnasa** [comment]
> I'm confused. Doesn't the library have it's own types?
Why do we need the outdated types package? 🤔


@emmenko 
hey you're right and I managed to get around it.
Thanks!

Closing this now.

I still think that `patch-package` is a dope tool, actually. so thanks for the tip, @tdeekens

**Pull request closed without merging**